src/GovUk.Education.ExploreEducationStatistics.Admin/ViewModels/Public.Data/BatchMappingUpdateViewModels.cs

```diff
@@ -0,0 +1,80 @@
+#nullable enable
+using System.Collections.Generic;
+using System.Linq;
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.ViewModels.Public.Data;
+
+public record BatchLocationMappingUpdatesRequest
+{
+    public List<LocationMappingUpdateRequest> Updates { get; init; } = [];
+
+    public class Validator : AbstractValidator<BatchLocationMappingUpdatesRequest>
+    {
+        public Validator()
+        {
+            RuleForEach(request => request.Updates)
+                .SetValidator(new LocationMappingUpdateRequest.Validator());
+        }
+    }
+}
+
+public record LocationMappingUpdateRequest
+{
+    public GeographicLevel? Level { get; init; }
+
+    public string? SourceKey { get; init; } = null;
```

Comment:
should this be non-nullable?

Reply:
Because I was forced to change the enums to be nullable so that they don't get auto-set, I did the same with these.  But actually there's no technical reason why they need to be, so I've set them back to being non-nullable to avoid future confusion!